test/jira-connection-test.js

```diff
@@ -8,4 +9,50 @@ describe('JIRA Connection Tests', function() {
         testJira.projectName.should.eql('test');
       });
   });
+
+  describe('findIssue Memoization', () => {
+    let jiraApi;
+    let spy;
+
+    let dummyJira = new DummyJira();
+
+    beforeEach(() => {
+      return getJiraAPI(path.join(process.cwd(), 'test', '.jirarc'))
+        .then(testJira => {
+          jiraApi = testJira;
+
+          jiraApi.request = sinon.stub(jiraApi, 'request', function(options, callback) {
+            let issueNumber = options.uri.split('/').pop().toString();
+            callback('', {statusCode: 200}, JSON.stringify(dummyJira.issues[issueNumber]));
+          });
+
+          spy = sinon.spy(jiraApi, 'doRequest');
+        });
+    });
+
+    it('findIssue with Same Key is Run Only Once', () => {
+      let firstResult;
+      return jiraApi.findIssue('WHP-9999')
+        .then(result => {
+          firstResult = result;
+          return jiraApi.findIssue('WHP-9999');
+        })
+        .then(result => {
+          assert.equal(spy.calledOnce, true);
+          assert.equal(result, firstResult);
+        });
```

Comment:
A cleaner way to do this:

```
return Promise.all([
    jiraApi.findIssue('WHP-9999'),
    jiraApi.findIssue('WHP-9999')
  ])
  .then(([first, second]) => {
    assert.equal(spy.calledOnce, true);
    assert.equal(result, firstResult);
  });
```

Utilize the array destructuring support in ES6.